### PR TITLE
Skip empty key variables in detect_aws_credentials

### PR DESCRIPTION
**Pull request by begoon**
This PR resolves the following - issue https://github.com/pre-commit/pre-commit-hooks/issues/449

### CODE REVIEW

File: pre_commit_hooks/detect_aws_credentials.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def check_file_for_aws_keys(
             for key in keys:
                 # naively match the entire file, low chance of incorrect
                 # collision
-                if key in text_body:
+                if key and key in text_body:
```
Comment (asottile) [inline pre_commit_hooks/detect_aws_credentials.py]: can you add a quick test for this to demonstrate the change -- definitely agree with this fix though :+1:

Comment (begoon) [reply pre_commit_hooks/detect_aws_credentials.py]: I moved the change to another place, and also added a test.

File: pre_commit_hooks/detect_aws_credentials.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ def get_aws_secrets_from_env() -> Set[str]:
     for env_var in (
         'AWS_SECRET_ACCESS_KEY', 'AWS_SECURITY_TOKEN', 'AWS_SESSION_TOKEN',
     ):
-        if env_var in os.environ:
+        if env_var in os.environ and os.environ[env_var]:
```
Comment (asottile) [inline pre_commit_hooks/detect_aws_credentials.py]: ~slightly simpler: `if os.environ.get(env_var)`

Comment (begoon) [reply pre_commit_hooks/detect_aws_credentials.py]: Fixed

**Comment by asottile** [comment]
thanks!

**Pull request merged**